mainhandler/handlerequests.go

```diff
@@ -233,15 +233,15 @@ func (mainHandler *MainHandler) HandleSingleRequest(ctx context.Context, session
 
 	actionHandler := NewActionHandler(mainHandler.clusterConfig, mainHandler.cfg, mainHandler.components, mainHandler.k8sAPI, sessionObj, mainHandler.commandResponseChannel, mainHandler.eventReceiverRestURL)
 	actionHandler.reporter.SetActionName(string(sessionObj.Command.CommandName))
-	actionHandler.reporter.SendDetails("Handling single request", true, sessionObj.ErrChan)
+	actionHandler.reporter.SendDetails("Handling single request", mainHandler.eventReceiverRestURL != "", sessionObj.ErrChan)
```

Comment:
Don't kill me, but I think the right approach here is to have a mock reporter so you initialize it when the server URL is not set.
What do you think?